code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -957,14 +966,20 @@ pub mod pallet {
 
 					let total_shares = T::Assets::total_issuance(rewards_pool.share_asset_id);
 
+					let pool_account = Self::pool_account_id(&stake.reward_pool_id);
+
 					for (reward_asset_id, reward) in &mut rewards_pool.rewards {
 						let new_inflation = if total_shares.is_zero() {
 							T::Balance::zero()
 						} else {
-							reward.total_rewards.safe_mul(&new_shares)?.safe_div(&total_shares)?
+							<T::Assets as orml_traits::MultiCurrency<_>>::free_balance(
```

Comment:
Does this work for native currency?

Reply:
it should since our assets pallet delegates to balances for native